### ISSUE TITLE
Errors are not passed through observable chain

### ISSUE THREAD

**Issue by Captnwalker1** [Nodonisko/ionic-cache#24]
So I have a service that wraps an http endpoint, and passes the observable back to a page. If an error happens the cache is preventing the error from getting passed to the page, and is only hitting the error handler in the service and not the page. It isn't even hitting the completion handler from the subscribe either, the error in the service seems to be stopping it.

If I return the request directly the errors happen as I would expect, service and page.
If I cache the request,  the service error is called but not the page error, even if there is no data already in the cache. I could see it not throwing the error if there was data in the cache.

I am not the most familiar with RxJs Observables so maybe this is an error on my part, but something still seems fishy with the cache not throwing the error.

Service:
```
loadAllDeals(): Observable<Deal[]> {
    let request = this.http.get(AppConfig.API_URL+'/users/'+this.user.auth_token()+'/deals')
      .map(res => <Deal[]>res.json())
      .catch(this.handleError);

    // return request;
    let response = this.cache.loadFromDelayedObservable(DealsService.cacheKey, request);
    return response;
  }

handleError(error) {
    console.log("DealsService handleError");
    return Observable.throw(error);
  }
```

Page:
```
loadDeals(locid?) {
    this.loading.present();
    this.DealsService.loadAllDeals()
      .catch(e => {return Observable.throw(e);})
      .subscribe(data => {
        this.deals = data;
      },
      (err) => {this.handleError(err);},
      () => {console.log("complete");}
    );
  }

handleError(error) {
    this.loading.dismiss();
    console.log("DealsPage handleError:");
    console.error(error);
    alert.present();
  }
```

**Comment by nkimaina** [comment]
@Captnwalker1 Did you find a solution for this? I could spare some time to create a PR

**Comment by Captnwalker1** [comment]
No solution, just a workaround used event system to fire an event in the service and listen for it in my page, so definitely still an issue.

**Comment by Nodonisko** [comment]
I think fix for this will be easy, but i don't know when i will have some free time.

**Comment by Nodonisko** [comment]
Solved in latest version #35
